### PR TITLE
make oscap tests run on rhel9 backend

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
Part of the oscap tests were too rhel8 backend specific and didn't run on rhel9 sat

### Solution
Making the tests more flexible

### Related Issues
Requires satelliteqe-jenkins mr 1300

Results form local rhel9 sat run:
```
pytest tests/foreman/cli/test_oscap.py 
...
============ 68 passed, 3 deselected, 35 warnings in 971.49s (0:16:11) =============

```

### CODE REVIEW

File: robottelo/constants/__init__.py [hunk 1]
```diff
@@ -2060,6 +2063,8 @@ class Colored(Box):
 EXPIRED_MANIFEST = 'expired-manifest.zip'
 
 # Data File Paths
+
+
```
Comment (Griffin-Sullivan) [inline robottelo/constants/__init__.py:2067]: Were these lines intended to be added?

**Review by Griffin-Sullivan — APPROVED** [review]
This should also fix our fapolicyd pipeline on RHEL9. Thanks for the quick fix

**Review by synkd — APPROVED** [review]

File: robottelo/constants/__init__.py (hunk 1 above)
Comment (pondrejk) [reply robottelo/constants/__init__.py]: my editor did this on save, made some config changes there :)

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6485
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_oscap.py --external-logging
Test Result : ========= 68 passed, 3 deselected, 187 warnings in 1152.37s (0:19:12) ==========
```

File: pytest_fixtures/component/oscap.py [hunk 2]
```diff
@@ -20,26 +20,32 @@ def tailoring_file_path(session_target_sat):
 
 
 @pytest.fixture(scope="session")
-def oscap_content_path(session_target_sat):
+def oscap_content_source(session_target_sat):
+    """Get scap content file path"""
+    return f'{settings.oscap.content_dir}/ssg-rhel{session_target_sat.os_distribution_version.split(".")[0]}-ds.xml'
```
Comment (jameerpathan111) [inline pytest_fixtures/component/oscap.py:25]:
```suggestion
    return f'{settings.oscap.content_dir}/ssg-rhel{session_target_sat.os_version.major}-ds.xml'
```
Same suggestion for all the other `os_distribution_version` occurrences.

**Review by jameerpathan111 — CHANGES_REQUESTED** [review]

**Comment by Griffin-Sullivan** [comment]
When you rerun PRT add `deploy_rhel_version: 9`

**Comment by jameerpathan111** [comment]
@pondrejk How about we update oscap.yaml in CI repo based on Satellite RHEL version? It should solve the issues for all the versions.

**Comment by pondrejk** [comment]
> @pondrejk How about we update oscap.yaml in CI repo based on Satellite RHEL version? It should solve the issues for all the versions. It should look something like:
> 
> ```
> OSCAP:
>   RHEL_MAJOR_VERSION: "@jinja {{this.server.version.rhel_version | int }}"
>   CONTENT_PATH: '@format /usr/share/xml/scap/ssg/content/ssg-rhel{this.oscap.rhel_major_version}-ds.xml'
>   # see: robottelo/constants/__init__.py OSCAP_PROFILE
>   PROFILE: '@format security{this.oscap.rhel_major_version}'
>   CONTENT_DIR: /usr/share/xml/scap/ssg/content
> ```

ok that seems superior, let me test it out

**Comment by pondrejk** [comment]
@jameerpathan111 @Griffin-Sullivan could you please take a look at satellite-jenkins mr !1316

**Comment by pondrejk** [comment]
Awaits satellite-jenkins config merge for prt run, local results with latest changes look good

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6649
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_oscap.py --external-logging
Test Result : ========= 68 passed, 3 deselected, 192 warnings in 1119.44s (0:18:39) ==========
```

File: pytest_fixtures/component/oscap.py [hunk 3]
```diff
@@ -35,11 +35,10 @@ def scap_content(import_ansible_roles, module_target_sat):
     )
     scap_id = scap_info['id']
     scap_info = entities.ScapContents(id=scap_id).read()
-
     scap_profile_id = [
         profile['id']
         for profile in scap_info.scap_content_profiles
-        if OSCAP_PROFILE[settings.oscap.profile] in profile['title']
+        if OSCAP_PROFILE[f'dsrhel{module_target_sat.os_version.major}'] in profile['title']
```
Comment (ogajduse) [inline pytest_fixtures/component/oscap.py:41]: A little suggestion here:
```suggestion
        if module_target_sat.OSCAP['dsrhel'] in profile['title']
```

see https://github.com/SatelliteQE/robottelo/blob/db5b3d5a3cffb8a827782b4ea4e979fb59c5a38a/robottelo/host_helpers/contenthost_mixins.py#L58-L64

A similar abstraction on the host level can be built for `OSCAP_DEFAULT_CONTENT`

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py

**Review by jameerpathan111 — APPROVED** [review]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6653
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_oscap.py --external-logging
Test Result : ========= 68 passed, 3 deselected, 188 warnings in 1176.13s (0:19:36) ==========
```

**Comment by pondrejk** [comment]
pre-commit.ci autofix

**Pull request merged**